src/main/java/Bosco.java

```diff
@@ -21,41 +22,70 @@ private static void printExitMessage() {
 
     public static void main(String[] args) {
         Scanner in = new Scanner(System.in);
-        String userInput;
+        String userInputString;
         String[] words;
         Task selectedTask;
 
         printWelcomeMessage();
 
         while (true) {
-            userInput = in.nextLine();
-            words = userInput.split(" ");
-            if (words[0].equals("bye")) {
+            userInputString = in.nextLine();
+            words = userInputString.split("\\s+", 2);
+            String commandType = words[0];
+            if (commandType.equals("bye")) {
                 break;
             }
+            String commandArgs = (words.length == 2 ? words[1] : words[0]);
+
             System.out.println(DIVIDER);
-            switch (words[0]) {
+            switch (commandType) {
             case "list":
                 for (int i = 0; i < taskCount; i++) {
                     System.out.println("\t " + (i + 1) + "." + tasks[i]);
                 }
                 break;
             case "mark":
-                selectedTask = tasks[Integer.parseInt(words[1]) - 1];
+                selectedTask = tasks[Integer.parseInt(commandArgs) - 1];
                 selectedTask.markAsDone();
                 System.out.println("\t Nice! I've marked this task as done:");
                 System.out.println("\t   " + selectedTask);
                 break;
             case "unmark":
-                selectedTask = tasks[Integer.parseInt(words[1]) - 1];
+                selectedTask = tasks[Integer.parseInt(commandArgs) - 1];
                 selectedTask.markAsNotDone();
                 System.out.println("\t OK, I've marked this task as not done yet:");
                 System.out.println("\t   " + selectedTask);
                 break;
```

Comment:
I think you may try to avoid having multiple levels of abstraction within a code fragment. SLAP will make the code more readable.

Reply:
Agreed, I will be adding more abstraction in future improvements.